### PR TITLE
Fix #79: Offer to install hg when not installed

### LINKED ISSUE #79: Instead of a bit confusing error when hg is not installed, offer to install hg.
When hg is not installed, you get `Command 'Hg: Clone' resulted in an error (command 'hg.clone' not found)`.

Instead, say that hg is not installed and offer to install it with a button.

### PR DESCRIPTION
**Pull request by hdpoliveira**

### CODE REVIEW

File: package.nls.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
 	"config.countBadge.all": "Include untracked files in count",
 	"config.countBadge.off": "No badge counter",
 	"config.allowPushNewBranches": "Whether new branches are pushed without warning",
+	"config.ignoreMissingHgWarning": "Ignores the warning when Hg is missing.",
```
Comment (incidentist) [inline package.nls.json:59]:
```suggestion
	"config.ignoreMissingHgWarning": "Ignores the warning when Mercurial is missing.",
```

File: src/hg.ts [hunk 2]
```diff
@@ -122,7 +122,8 @@ export class HgFinder {
 				case 'win32': return this.findHgWin32();
 				default: return this.findSpecificHg('hg');
 			}
-		});
+		})
+		.then(null, () => Promise.reject(new Error('Hg installation not found.')));
```
Comment (incidentist) [inline src/hg.ts:126]:
```suggestion
		.then(null, () => Promise.reject(new Error('Mercurial installation not found.')));
```

File: src/main.ts [hunk 3]
```diff
@@ -6,51 +6,115 @@
...
+
+	try {
+		const dotHgStat = await new Promise<fs.Stats>((c, e) => fs.stat(dotHg, (err, stat) => err ? e(err) : c(stat)));
+		return dotHgStat.isDirectory();
+	} catch (err) {
+		return false;
+	}
+}
+
+async function warnAboutMissingHg(): Promise<void> {
```
Comment (incidentist) [inline src/main.ts:36]: This function should be in `interaction.ts`, which is where this extension keeps stuff related to UI. This is different from how the Git extension does it.

File: src/main.ts [hunk 4]
```diff
@@ -6,51 +6,115 @@
...
+	const areHgRepositories = await Promise.all(workspace.workspaceFolders.map(isHgRepository));
+
+	if (areHgRepositories.every(isHgRepository => !isHgRepository)) {
+		return;
+	}
+
+	const download = localize('downloadhg', "Download Hg");
+	const neverShowAgain = localize('neverShowAgain', "Don't Show Again");
+	const choice = await window.showWarningMessage(
+		localize('notfound', "Hg not found. Install it or configure it using the 'hg.path' setting."),
```
Comment (incidentist) [inline src/main.ts:57]:
```suggestion
		localize('notfound', "Mercurial was not found. Install it or configure it using the 'hg.path' setting."),
```

File: src/main.ts [hunk 5]
```diff
@@ -6,51 +6,115 @@
...
+		hg.onOutput(str => outputChannel.append(str), null, disposables);
+	
+		disposables.push(
+			new CommandCenter(hg, model, outputChannel),
+			new HgContentProvider(model),
+		);
+	
+		await checkHgVersion(info);
+	} catch (err) {
+		if (!/Hg installation not found/.test(err.message || '')) {
```
Comment (incidentist) [inline src/main.ts:108]:
```suggestion
		if (!/Mercurial installation not found/.test(err.message || '')) {
```

**Comment by hdpoliveira** [comment]
The following warning is shown when user opens a folder/workspace in which there is a .hg directory, but Mercurial installation is not found:

image

Clicking the Download Mercurial goes to https://www.mercurial-scm.org/
Clicking the Don't Show Again sets hg.ignoreMissingHgWarning

**Pull request merged**